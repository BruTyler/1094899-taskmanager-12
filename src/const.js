export const Color = {
  BLACK: `black`,
  YELLOW: `yellow`,
  BLUE: `blue`,
  GREEN: `green`,
  PINK: `pink`,
};
