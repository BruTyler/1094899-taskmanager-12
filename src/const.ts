export enum Color {
  BLACK = `black`,
  YELLOW = `yellow`,
  BLUE = `blue`,
  GREEN = `green`,
  PINK = `pink`,
}

export enum RenderPosition {
  AFTERBEGIN = `afterbegin`,
  BEFOREEND = `beforeend`
}

export enum SortType {
  DEFAULT = `DEFAULT`,
  DATE_UP = `DATE up`,
  DATE_DOWN = `DATE down`
}

export enum RepeatingDaysMap {
  mo = 1,
  tu = 2,
  we = 4,
  th = 8,
  fr = 16,
  sa = 32,
  su = 64
}
