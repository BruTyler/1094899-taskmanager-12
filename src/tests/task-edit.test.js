import TaskEditView from '../view/task-edit.js';
import {Color} from '../const.js';

it(`TaskEditor rendering`, () => {
  const task = {
    description: `some description1`,
    dueDate: new Date(2020, 8, 4),
    repeatingDays: {
      mo: false,
      tu: false,
      we: false,
      th: false,
      fr: false,
      sa: false,
      su: true
    },
    color: Color.BLACK,
    isFavorite: true,
    isArchive: false,
  };

  const generatedTree = new TaskEditView(task).getTemplate();

  expect(generatedTree).toMatchSnapshot();
});
