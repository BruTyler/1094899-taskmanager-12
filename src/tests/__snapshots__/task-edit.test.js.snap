// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskEditor rendering 1`] = `
"<article class=\\"card card--edit card--black  card--repeat\\">
    <form class=\\"card__form\\" method=\\"get\\">
      <div class=\\"card__inner\\">
        <div class=\\"card__color-bar\\">
          <svg class=\\"card__color-bar-wave\\" width=\\"100%\\" height=\\"10\\">
            <use xlink:href=\\"#wave\\"></use>
          </svg>
        </div>
        <div class=\\"card__textarea-wrap\\">
          <label>
            <textarea
              class=\\"card__text\\"
              placeholder=\\"Start typing your text here...\\"
              name=\\"text\\"
            >some description1</textarea>
          </label>
        </div>
        <div class=\\"card__settings\\">
          <div class=\\"card__details\\">
            <div class=\\"card__dates\\">
              <button class=\\"card__date-deadline-toggle\\" type=\\"button\\">
      date: <span class=\\"card__date-status\\">yes</span>
    </button>
    <fieldset class=\\"card__date-deadline\\">
      <label class=\\"card__input-deadline-wrap\\">
        <input
          class=\\"card__date\\"
          type=\\"text\\"
          placeholder=\\"\\"
          name=\\"date\\"
          value=\\"September 4\\"
        />
      </label>
    </fieldset>
  
              <button class=\\"card__repeat-toggle\\" type=\\"button\\">
    repeat:<span class=\\"card__repeat-status\\">yes</span>
  </button>
  <fieldset class=\\"card__repeat-days\\">
        <div class=\\"card__repeat-days-inner\\">
          <input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-mo\\"
            name=\\"repeat\\"
            value=\\"mo\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-mo\\"
            >mo</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-tu\\"
            name=\\"repeat\\"
            value=\\"tu\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-tu\\"
            >tu</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-we\\"
            name=\\"repeat\\"
            value=\\"we\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-we\\"
            >we</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-th\\"
            name=\\"repeat\\"
            value=\\"th\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-th\\"
            >th</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-fr\\"
            name=\\"repeat\\"
            value=\\"fr\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-fr\\"
            >fr</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-sa\\"
            name=\\"repeat\\"
            value=\\"sa\\"
            
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-sa\\"
            >sa</label
          ><input
            class=\\"visually-hidden card__repeat-day-input\\"
            type=\\"checkbox\\"
            id=\\"repeat-su\\"
            name=\\"repeat\\"
            value=\\"su\\"
            checked
          />
          <label class=\\"card__repeat-day\\" for=\\"repeat-su\\"
            >su</label
          >
        </div>
      </fieldset>
            </div>
          </div>
          <div class=\\"card__colors-inner\\">
            <h3 class=\\"card__colors-title\\">Color</h3>
            <div class=\\"card__colors-wrap\\">
              <input
    type=\\"radio\\"
    id=\\"color-black\\"
    class=\\"card__color-input card__color-input--black visually-hidden\\"
    name=\\"color\\"
    value=\\"black\\"
    checked
  />
  <label
    for=\\"color-black\\"
    class=\\"card__color card__color--black\\"
    >black</label
  ><input
    type=\\"radio\\"
    id=\\"color-yellow\\"
    class=\\"card__color-input card__color-input--yellow visually-hidden\\"
    name=\\"color\\"
    value=\\"yellow\\"
    
  />
  <label
    for=\\"color-yellow\\"
    class=\\"card__color card__color--yellow\\"
    >yellow</label
  ><input
    type=\\"radio\\"
    id=\\"color-blue\\"
    class=\\"card__color-input card__color-input--blue visually-hidden\\"
    name=\\"color\\"
    value=\\"blue\\"
    
  />
  <label
    for=\\"color-blue\\"
    class=\\"card__color card__color--blue\\"
    >blue</label
  ><input
    type=\\"radio\\"
    id=\\"color-green\\"
    class=\\"card__color-input card__color-input--green visually-hidden\\"
    name=\\"color\\"
    value=\\"green\\"
    
  />
  <label
    for=\\"color-green\\"
    class=\\"card__color card__color--green\\"
    >green</label
  ><input
    type=\\"radio\\"
    id=\\"color-pink\\"
    class=\\"card__color-input card__color-input--pink visually-hidden\\"
    name=\\"color\\"
    value=\\"pink\\"
    
  />
  <label
    for=\\"color-pink\\"
    class=\\"card__color card__color--pink\\"
    >pink</label
  >
            </div>
          </div>
        </div>
        <div class=\\"card__status-btns\\">
          <button class=\\"card__save\\" type=\\"submit\\">save</button>
          <button class=\\"card__delete\\" type=\\"button\\">delete</button>
        </div>
      </div>
    </form>
  </article>"
`;
